src/clp_ffi_py/Python.hpp

```diff
@@ -8,3 +8,20 @@
 #define PY_SSIZE_T_CLEAN
 #include <Python.h>
 #include <structmember.h>
+
+// clang-format off
+#ifdef CLP_FFI_PY_ENABLE_LINTING
+// The following headers are added to export Python headers to get rid of clang-tidy warnings.
+// IWYU pragma: begin_exports
+#include <bytesobject.h>
+#include <dictobject.h>
+#include <methodobject.h>
+#include <modsupport.h>
+#include <object.h>
+#include <objimpl.h>
+#include <pyerrors.h>
+#include <pymacro.h>
+#include <typeslots.h>
```

Comment:
I haven't verified that these are all necessary. Let's do that in #76.

Reply:
I'm ok to defer them to other PRs, but just wanna make an agreement that this (IWYU + macro) will be our solution to resolve clang-tidy warnings

Reply:
As we discussed offline, I'm not suggesting we remove this from this PR. I'm saying we can review it in #76.